precompiles/regular/regular.go

```diff
@@ -0,0 +1,262 @@
+	return ABI
+}
+
+// RequiredGas calculates the contract gas use
+func (rc *Contract) RequiredGas(input []byte) uint64 {
+	// base cost to prevent large input size
+	baseCost := uint64(len(input)) * rc.kvGasConfig.WriteCostPerByte
+
+	// get methodID (first 4 bytes)
+	var methodID [4]byte
+	copy(methodID[:], input[:4])
+
+	if requiredGas, ok := GasRequiredByMethod[methodID]; ok {
+		return requiredGas + baseCost
+	}
+
+	return baseCost
+}
+
+func (rc *Contract) Bech32ToHexAddr(method *abi.Method, args []interface{}) ([]byte, error) {
+	if len(args) != 1 {
+		return nil, &ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 1,
+		}
+	}
+
+	address, ok := args[0].(string)
+	if !ok || address == "" {
+		return nil, fmt.Errorf("invalid bech32 address: %v", args[0])
+	}
+
+	bech32Prefix := strings.SplitN(address, "1", 2)[0]
+	if bech32Prefix == address {
+		return nil, fmt.Errorf("invalid bech32 address: %s", address)
+	}
+
+	addressBz, err := sdk.GetFromBech32(address, bech32Prefix)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := sdk.VerifyAddressFormat(addressBz); err != nil {
+		return nil, err
+	}
+
+	return method.Outputs.Pack(common.BytesToAddress(addressBz))
+}
+
+func (rc *Contract) Bech32ify(method *abi.Method, args []interface{}) ([]byte, error) {
+	if len(args) != 2 {
+		return nil, &ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 2,
+		}
+	}
+
+	cfg := sdk.GetConfig()
+	prefix, _ := args[0].(string)
+	if strings.TrimSpace(prefix) == "" {
+		return nil, fmt.Errorf(
+			"invalid bech32 human readable prefix (HRP). Please provide a either an account, validator or consensus address prefix (eg: %s, %s, %s)",
+			cfg.GetBech32AccountAddrPrefix(),
+			cfg.GetBech32ValidatorAddrPrefix(),
+			cfg.GetBech32ConsensusAddrPrefix(),
+		)
+	}
+
+	address, ok := args[1].(common.Address)
+	if !ok {
+		return nil, fmt.Errorf("invalid hex address")
+	}
+
+	// NOTE: safety check, should not happen given that the address is is 20 bytes.
+	if err := sdk.VerifyAddressFormat(address.Bytes()); err != nil {
+		return nil, err
+	}
+
+	bech32Str, err := sdk.Bech32ifyAddressBytes(prefix, address.Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	addressBz, err := sdk.GetFromBech32(bech32Str, "zeta")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := sdk.VerifyAddressFormat(addressBz); err != nil {
+		return nil, err
+	}
+
+	return method.Outputs.Pack(bech32Str)
+}
+
+func (rc *Contract) RegularCall(ctx sdk.Context, method *abi.Method, args []interface{}) ([]byte, error) {
+	if len(args) != 2 {
+		return nil, &(ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 2,
+		})
+	}
+	callingMethod, ok := args[0].(string)
+	if !ok {
+		return nil, &ptypes.ErrInvalidMethod{
+			Method: callingMethod,
+		}
+	}
+	callingContract, ok := args[1].(common.Address)
+	if !ok {
+		return nil, ptypes.ErrInvalidAddr{
+			Got: callingContract.String(),
+		}
+	}
+
+	res, err := rc.FungibleKeeper.CallEVM(
```

Comment:
Even if it is just for test purpose, I would consider this example a bit dangerous,  the fungible module can mint ZRC20, meaning this precompiled could be used to mint infinite amount of ZRC20s

Even disable I would avoid having it defined in our codebase 😄

I would change it to a more scope and simple use case:

One thing I have in mind is interacting with the gas stability pool. In the fungible module `gas_stability_pool.go` provides pools autofilled from cctx fees that are used to fund cctx when gas price hikes. Eventually they can be manually funded by users.
You have a method to get the address of the gas stability pool from a chain ID:
```
func (k Keeper) GetGasStabilityPoolBalance(
	ctx sdk.Context,
	chainID int64,
)
```
We could have a precompile allow to get this address from a smart contract call.

Reply:
Makes sense, I'll modify the contract.